lints/lint_dnsname_wildcard_left_of_public_suffix.go

```diff
@@ -0,0 +1,54 @@
+package lints
+
+import (
+	"github.com/weppos/publicsuffix-go/publicsuffix"
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type DNSNameWildcardLeftofPublicSuffix struct {
+	// Internal data here
+}
+
+func (l *DNSNameWildcardLeftofPublicSuffix) Initialize() error {
+	return nil
+}
+
+func (l *DNSNameWildcardLeftofPublicSuffix) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubscriberCert(c)
+}
+
+func wildcardLeftOfPublicSuffix(domain string) bool {
+	parsedDomain, err := publicsuffix.Parse(domain)
+	if err != nil {
+		return false
+	}
+	if parsedDomain.TRD == "" {
```

Comment:
why is this a check? Having `terrible.*.com` should still fail this lint.

Reply:
This test checks _only_ for the case *.<ETLD>, for example, *.co.uk, or *.com.

From the BRs: "the CA MUST establish and follow a documented procedure[^pubsuffix] that determines if the wildcard character occurs in the first label position to the left of a “registry‐controlled” label or “public suffix”".

Since we can't explicitly test the "documented procedure", @titanous recommended we do this as a warning in case we see this in the wild.

Reply:
(the other wildcard tests were merged last friday)